ecosystem/typescript/sdk/src/token_client.ts

```diff
@@ -47,18 +43,31 @@ export class TokenClient {
     description: string,
     uri: string,
   ): Promise<Types.HexEncodedBytes> {
-    const payload: Types.TransactionPayload = {
-      type: 'script_function_payload',
-      function: '0x1::Token::create_unlimited_collection_script',
-      type_arguments: [],
-      arguments: [
-        Buffer.from(name).toString('hex'),
-        Buffer.from(description).toString('hex'),
-        Buffer.from(uri).toString('hex'),
-      ],
-    };
-    const transactionHash = await this.submitTransactionHelper(account, payload);
-    return transactionHash;
+    const payload = new TxnBuilderTypes.TransactionPayloadScriptFunction(
+      TxnBuilderTypes.ScriptFunction.natural(
```

Comment:
This is cool. try to learn more. Does this get updated as they recompile the move scripts?